config/deploy.rb

```diff
@@ -105,7 +105,8 @@ class PrecompileRequired < StandardError;
   desc "Copy config files from <deploy_to>/config to the release directory"
   task :copy_config do
     on roles(:app) do
-      execute :sudo, 'cp -rp', "#{fetch(:deploy_to)}/config/*", "#{release_path}/config"
+      execute :sudo, 'rsync -avr --exclude='*.env*', "#{fetch(:deploy_to)}/config/*", "#{release_path}/config"
+      execute :sudo, 'cp -p', "#{fetch(:deploy_to)}/config/.env.local", "#{release_path}"
```

Comment:
So if my understanding is correct....

Here we are excluding the `.env` files from being copied from the deploy location to the release location so that they both keep their own values with the exception of `.env.local` which is copied individually  (with it's parent paths) to the release location?

Reply:
Yup, that's right. Basically, the reason that I'm excluding all other `.env` files from being copied over is to keep things consistent. When/if we check into src ctrl shared `.env` files that every environment (staging or prod) needs and which are not secret, I don't want to encourage people to customize things locally by dropping in a new version of that checked in `.env` file into the deploy config dir. I want it to be very explicit that all local config goes in the `.env.local` file if we want to override configs with local behavior.

Reply:
Also, all this is temporary until we can cutover to Heroku. At that point, local `.env` files and how we handle them won't matter as much. We just need to make sure that for each app, we set the variables in Heroku for that app. I'll give some more thought on how to handle that and make it easy for deploys and devs to manage once we're at the point of being ready to make the switch from Capistrano to Heroku though.